specs/commandline/blob.md

```diff
@@ -0,0 +1,363 @@
+            "scopes": [ "blob:*" ],
+            "signatureVerification": {
+              "level" : "audit"
+            },
+            "trustStores": ["ca:acme-rockets"],
+            "trustedIdentities": ["*"]
+        }
+    ]
+}
+```
+
+### Verify the detached signature of a blob
+
+Configure trust store and trust policy properly before using `notation blob verify` command.
+
+```shell
+
+# Prerequisites: Signature is produced on the filesystem from `notation blob sign` command.
+# Configure trust store by adding a certificate file into trust store named "wabbit-network" of type "ca"
+notation certificate add --type ca --store wabbit-networks wabbit-networks.crt
+
+# Create a JSON file named "trustpolicy.json" under directory "{NOTATION_CONFIG}".
+
+# Verify the detached signature
+notation blob verify --signature /tmp/my-blob-signature.sig.jws /tmp/my-blob.bin
+```
+
+An example of output messages for a successful verification:
+
+```text
+Successfully verified signature /tmp/my-blob-signature.sig.jws
+```
+
+### Verify the signature with user metadata
+
+Use the `--user-metadata` flag to verify that provided key-value pairs are present in the payload of the valid signature.
+
+```shell
+# Verify the signature and verify that io.wabbit-networks.buildId=123 is present in the signed payload
+notation blob verify --user-metadata io.wabbit-networks.buildId=123 --signature /tmp/my-blob-signature.sig.jws /tmp/my-blob.bin
+```
+
+An example of output messages for a successful verification:
+
+```text
+Successfully verified signature /tmp/my-blob-signature.sig.jws
+
+The blob signature is having the following user metadata.
+
+KEY                         VALUE
+io.wabbit-networks.buildId  123
+```
+
+An example of output messages for an unsuccessful verification:
+
+```text
+Error: signature verification failed: unable to find specified metadata in the given signature
+```
+
+### Verify the signature with media type
+
+Use the `--media-type` flag to verify that signature is for the provided media-type.
+
+```shell
+# Verify the signature and verify that io.wabbit-networks.buildId=123 is present in the signed payload
+notation blob verify --media-type application/my-media-octet-stream --signature /tmp/my-blob-signature.sig.jws /tmp/my-blob.bin
+```
+
+An example of output messages for a successful verification:
+
+```text
+Successfully verified signature /tmp/my-blob-signature.sig.jws
+
+The blob is of media type `application/my-media-octet-stream`.
+
+```
+
+An example of output messages for an unsuccessful verification:
+
+```text
+Error: signature verification failed: The blob is not of media type `application/my-media-octet-stream`.
+```
+
+### Verify the signature using a policy scope
+
+Use the `--policy-scope` flag to select a Policy scope to verify the signature against.
+
+```shell
+notation blob verify --policy-scope my-blob-verification-selector --signature /tmp/my-blob-signature.sig.jws /tmp/my-blob.bin
```

Comment:
For consistency, maybe you can consider using the scope from the above policy example. my-blob-verification-selector doesn't exist.

Reply:
Good call.